mods/Fifty.ServerTeamBalance/mod/scripts/vscripts/fstb_command_callbacks.nut

```diff
@@ -12,33 +12,35 @@ void function FSTM_CommandCallback_Switch( entity player, array< string > args )
 		return
 	}
 
-	entity target = player
-
+	entity target
 	if( FSA_IsAdmin( player ) && args.len() != 0 ) {
-		bool foundTarget = false
 		foreach( entity p in GetPlayerArray() ) {
 			if( p.GetPlayerName().tolower().find( args[0].tolower() ) != -1 ) {
+				if( target != null ){
+					FSU_PrivateChatMessage(player, "%EMore than one matching player! %TWrite a bit more of their name.")
+					return
+				}
 				target = p
-				foundTarget = true
-				break
 			}
 		}
-
-		if( !foundTarget ) {
+		if(target == null){
 			FSU_PrivateChatMessage( player, "%H\"" + args[0] + "\"%E couldn't be found!" )
 			return
 		}
+		else
+			FSU_PrivateChatMessage( player, "%S The team of %H" + target.GetPlayerName() + "%S has been switched!" )
 	}
+	else
+		target = player
 
 	if( !IsValid( target ) ) {
-		FSU_PrivateChatMessage( player, "Player entity isn't valid!" )
+		FSU_PrivateChatMessage( player, "%EThat player entity isn't valid!" )
 		return
 	}
 
-	if( IsAlive( player ) )
+	if( IsAlive(player) && player == target )
 		player.Die()
```

Comment:
We want to kill the target, not the player calling this, explained in the previous comment

Reply:
No, we don't.

This is deliberate. When the target is different from player, killing them is unnecessary.

Reply:
We kill targets so that they respawn with their team. There's also the issue of some ui not updating when you switch teams without respawning.